libc/resolve_path.c

```diff
@@ -73,7 +73,11 @@ TEST(resolve_path, canonicalize_abs_simple)
 
 TEST(resolve_path, canonicalize_pwd_simple)
 {
-	/* don't care about current dir */
+	char twd[PATH_MAX];
+
+	/* save the test working diectory */
+	TEST_ASSERT_NOT_NULL(getcwd(twd, sizeof(twd)));
```

Comment:
as most of the tests do change the cwd, we could make saving/restoring cwd in  `TEST_SETUP`/`TEST_TEAR_DOWN`.

I would appreciate more descriptive variable name than `twd`.

Reply:
I will do it this way for both `unistd_fsdir` and `resolve_path`, even if most of cases do not need it, I think it will be safer and even if some case failed it would go back to `testWorkDir`